zerver/tests/test_auth_backends.py

```diff
@@ -784,8 +784,14 @@ def social_auth_test(self, account_data_dict: Dict[str, str],
                 # it's definitely the GitHub authentication backend.
                 if self.AUTH_FINISH_URL == "/complete/github/":
                     self.assert_in_success_response(["Select account"], result)
-                    result = self.client_get(self.AUTH_FINISH_URL,
-                                             dict(state=csrf_state, email=account_data_dict['email']), **headers)
+                    if account_data_dict.get('avatar_url'):
+                        user_data = dict(state=csrf_state,
+                                         email=account_data_dict['email'],
+                                         avatar_url=account_data_dict.get('avatar_url'))
+                    else:
+                        user_data = dict(state=csrf_state, email=account_data_dict['email'])
```

Comment:
Why do we need the if/else thing, can't we do
```
                        user_data = dict(state=csrf_state,
                                         email=account_data_dict['email'],
                                         avatar_url=account_data_dict.get('avatar_url'))
```
in both cases?

Reply:
I was letting the user_avatar_url on that dict but when I ran all tests in test_auth_backends.py, one of the test_github_oauth2 broke because the user_avatar_url was None as it was not passing the URL. So, I was a bit afraid to change another test as it was testing something else, other than github_avatar_url.

Reply:
But should I change the other test?

Reply:
Ah, I see what you mean. Okay, so the core use of the account_data_dict is here in `social_auth_test`:
```
            requests_mock.add(
                requests_mock.GET,
                self.USER_INFO_URL,
                status=200,
                body=json.dumps(account_data_dict)
            )
```
So this mock wants to mimic a (simplified) response that this github API endpoint returns in real life. So if it returns `avatar_url` (or some other name) in the json response, then I think the best approach is to mimic having that always. In cases where we're not testing the avatar upload logic (so in all pre-existing tests?), this value in the `account_data_dict` should match what the github endpoint returns for users with no avatar. Probably it's `None`? Or empty string? This should be checked and then all the existing tests modified to have these dicts fit that convention. That makes these tests closest to simulating the real data the code will be running again.

Reply:
I couldn't find the values avatar_url could have on the Github API documentation, but for I tested with a profile with no avatar and no gravatar associated, and there was a URL to the default avatar, also with no avatar uploaded from the user and a gravatar associated, there was always a string on avatar_url.

Reply:
But I saw that when there is an attribute that is a string and has no value, the default is an empty string.

Reply:
Oh, I see! We should do empty string then I suppose?